File: data/src/mock/java/com/omvp/data/StaticRepository.java
package com.omvp.data;

import android.net.Uri;

import com.omvp.domain.SampleDomain;

import org.threeten.bp.LocalDateTime;

import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.UUID;

import timber.log.Timber;

public class StaticRepository {

    private static final int NUM_ITEMS = 20;

    public static Map<String, SampleDomain> sampleDomainList = new LinkedHashMap<>(NUM_ITEMS);

    public static void init() {
        if (sampleDomainList.isEmpty()) {
            for (int i = 1; i <= NUM_ITEMS; i++) {
                String id = UUID.randomUUID().toString();
                sampleDomainList.put(id, initSampleDomain(id, i));
            }
        }
    }

    private static SampleDomain initSampleDomain(String id, int position) {
        return new SampleDomain(
                id,
                "item " + position,
                Uri.parse("https://www.google.com"),
                LocalDateTime.now(),
                R.mipmap.ic_launcher_round
        );
    }

}

File: data/src/mock/java/com/omvp/data/StaticRepository.kt
package com.omvp.data

import android.net.Uri
import com.omvp.domain.SampleDomain
import org.threeten.bp.LocalDateTime
import java.util.*

object StaticRepository {

    private const val NUM_ITEMS = 20

    var sampleDomainList: MutableMap<String, SampleDomain> = LinkedHashMap(NUM_ITEMS)

    fun init() {
        if (sampleDomainList.isEmpty()) {
            for (i in 1..NUM_ITEMS) {
                val id = UUID.randomUUID().toString()
                sampleDomainList[id] = initSampleDomain(id, i)
            }
        }
    }

    private fun initSampleDomain(id: String, position: Int): SampleDomain {
        return SampleDomain(
                id,
                "item $position",
                Uri.parse("https://www.google.com"),
                LocalDateTime.now(),
                R.mipmap.ic_launcher_round
        )
    }

}
